src/app/codegen-data-model/CodegenDataModel_Write.cpp

```diff
@@ -0,0 +1,341 @@
+    case ZCL_INT32U_ATTRIBUTE_TYPE: // Unsigned 32-bit integer
+        return DecodeIntoEmberBuffer<uint32_t>(decoder, isNullable, out);
+    case ZCL_INT40U_ATTRIBUTE_TYPE: // Unsigned 40-bit integer
+        return DecodeIntoEmberBuffer<OddSizedInteger<5, false>>(decoder, isNullable, out);
+    case ZCL_INT48U_ATTRIBUTE_TYPE: // Unsigned 48-bit integer
+        return DecodeIntoEmberBuffer<OddSizedInteger<6, false>>(decoder, isNullable, out);
+    case ZCL_INT56U_ATTRIBUTE_TYPE: // Unsigned 56-bit integer
+        return DecodeIntoEmberBuffer<OddSizedInteger<7, false>>(decoder, isNullable, out);
+    case ZCL_INT64U_ATTRIBUTE_TYPE: // Unsigned 64-bit integer
+        return DecodeIntoEmberBuffer<uint64_t>(decoder, isNullable, out);
+    case ZCL_INT8S_ATTRIBUTE_TYPE: // Signed 8-bit integer
+        return DecodeIntoEmberBuffer<int8_t>(decoder, isNullable, out);
+    case ZCL_INT16S_ATTRIBUTE_TYPE: // Signed 16-bit integer
+        return DecodeIntoEmberBuffer<int16_t>(decoder, isNullable, out);
+    case ZCL_INT24S_ATTRIBUTE_TYPE: // Signed 24-bit integer
+        return DecodeIntoEmberBuffer<OddSizedInteger<3, true>>(decoder, isNullable, out);
+    case ZCL_INT32S_ATTRIBUTE_TYPE: // Signed 32-bit integer
+        return DecodeIntoEmberBuffer<int32_t>(decoder, isNullable, out);
+    case ZCL_INT40S_ATTRIBUTE_TYPE: // Signed 40-bit integer
+        return DecodeIntoEmberBuffer<OddSizedInteger<5, true>>(decoder, isNullable, out);
+    case ZCL_INT48S_ATTRIBUTE_TYPE: // Signed 48-bit integer
+        return DecodeIntoEmberBuffer<OddSizedInteger<6, true>>(decoder, isNullable, out);
+    case ZCL_INT56S_ATTRIBUTE_TYPE: // Signed 56-bit integer
+        return DecodeIntoEmberBuffer<OddSizedInteger<7, true>>(decoder, isNullable, out);
+    case ZCL_INT64S_ATTRIBUTE_TYPE: // Signed 64-bit integer
+        return DecodeIntoEmberBuffer<int64_t>(decoder, isNullable, out);
+    case ZCL_SINGLE_ATTRIBUTE_TYPE: // 32-bit float
+        return DecodeIntoEmberBuffer<float>(decoder, isNullable, out);
+    case ZCL_DOUBLE_ATTRIBUTE_TYPE: // 64-bit float
+        return DecodeIntoEmberBuffer<double>(decoder, isNullable, out);
+    case ZCL_CHAR_STRING_ATTRIBUTE_TYPE: // Char string
+        return DecodeStringLikeIntoEmberBuffer<CharSpan, ShortPascalString>(decoder, isNullable, out);
+    case ZCL_LONG_CHAR_STRING_ATTRIBUTE_TYPE:
+        return DecodeStringLikeIntoEmberBuffer<CharSpan, LongPascalString>(decoder, isNullable, out);
+    case ZCL_OCTET_STRING_ATTRIBUTE_TYPE: // Octet string
+        return DecodeStringLikeIntoEmberBuffer<ByteSpan, ShortPascalString>(decoder, isNullable, out);
+    case ZCL_LONG_OCTET_STRING_ATTRIBUTE_TYPE:
+        return DecodeStringLikeIntoEmberBuffer<ByteSpan, LongPascalString>(decoder, isNullable, out);
+    default:
+        ChipLogError(DataManagement, "Attribute type 0x%x not handled", static_cast<int>(metadata->attributeType));
+        return CHIP_IM_GLOBAL_STATUS(UnsupportedWrite);
```

Comment:
This is the wrong value --> `UnsupportedWrite` can only be used when a write is attempted to a read-only value. Please revisit every usage you make of UnsupportedWrite, and replace with ConstraintError, most likely.